update read_health_facilities on python

vss-2 opened the pull request:
Implements the same feature present on R package and issued in #342 .

rafapereirabr commented:
Thanks @vss-2 . Your contribution is very much appreciated! The PR seems to failing all the tests, though. Could please have a look at this?

vss-2 commented:
Sure! The tests were failing because this first commit was sent before I send the other PR fixing the python tests pipeline. 
I had to merge with main branch to update this issue. 
There was also another error: I forgot to update expected dataset size on the python test. 
I've fixed this, added the "banana" tests cases, also updated the error message to "year/date" on utils, following R version standard.

vss-2 commented:
Sorry, after tests didn't finishing well I noticed that adding a case where "verbose=banana" is acceptable in Python conditionals, yikes!
In R, the "banana" attribution to "showProgress" throws an exception, so tests aren't all that similar now hahahah.
